### PR TITLE
add example for getting the timezone from location

### LINKED ISSUE #676: Feature request: convert longitude values to timezones
In an Astrodynamics course I am following I had the need to do some operations on times and dates: handle time zones, find out sidereal times, position observers according to time measurements, etc. One of the key operations I have to perform is: given the longitude (we usually neglect irregular timezones, hence no need for latitude) of an observer, find out its time zone (also given a particular timezone, find out the longitude of the central meridian). Anyone else thinks Astropy would be suitable for this? I found some web services with this purpose, but having this functionality in this Python package would be ideal. I think it is also available in a package named astral (http://pypi.python.org/pypi/astral/).

EDIT: the `good first issue` refers to the `docs` part of this issue - what would be needed is to find where best to put the example given in https://github.com/astropy/astropy/issues/676#issuecomment-623138161

### PR DESCRIPTION
**Pull request by jbmorgado**
### Description

This pull request is to address the lack of an example in documentation, on how to find out
the local timezone for an observer given his position.

It uses the `zoneinfo` standard module from python. This module is provided by default on python >=3.9,
the example adds an workaround for python<3.9 using the `backports` package.

Fixes #676

### CODE REVIEW

File: docs/coordinates/timezones.rst
Comment (pllim) [inline docs/coordinates/timezones.rst:27]: We declare dependencies in `setup.cfg`.

**Review by pllim — CHANGES_REQUESTED** [review]
I think this file needs to be listed in a TOC somewhere? Or does it really have to be a separate file or can it be a new section in an existing file instead? Or maybe this is better suited for `astropy-tutorials`?

I'll let subpackage maintainer decide on the decision to introduce a new optional dependency for this documentation.

Need some changes to follow https://docs.astropy.org/en/latest/development/style-guide.html .

I don't think `doctest` is running this without the `>>>`  prompt. `doctest` should run, right?

File: docs/coordinates/timezones.rst
Comment (jbmorgado) [reply docs/coordinates/timezones.rst]: Thank you for the feedback.

I added the `>>>` prompt (although it impairs readability, are we supposed to use those everywhere?) and removed the explicit pip install of `backports`.

I don't know about the TOC. Maybe someone else should decide where this goes.

I also went through the style guide, by my poor grammar knowledge doesn't allow me to see where the style errors lie on the regard

**Comment by jbmorgado** [comment]
The CI pipeline is failing at unexpected points. This all run in the previous commit and all I did was to change some lines in the documentation: they shouldn't have any impact on those tests.

**Comment by pllim** [comment]
The failure is indeed related. You import `timezonefinder`, which is not part of Python standard library nor any existing dependency.

**Comment by pllim** [comment]
I am started to think perhaps this is not a suitable example for core lib since this one short example introduces two new dependencies but other maintainers may think differently.

**Comment by jbmorgado** [comment]
True, I missed that timezonefinder wasn't a part of astropy requirements as well.

**Comment by astrojuanlu** [comment]
I am not sure the extra two dependencies are worth adding either. Left a comment https://github.com/astropy/astropy/issues/676#issuecomment-748587680 about this.

**Comment by pllim** [comment]
Okay then, I added a "Close?" label, which means the bot will close this PR without merge the next time it runs unless there are objections. Let's continue the discussion over at #676 if desired (until it is also closed by the bot).

@jbmorgado , we still thank you very much for this PR even if we don't end up merging it, as it has provided much needed insight to move the discussions forward. We hope you would continue to contribute in the future! 🙇‍♀️

**Comment by adrn** [comment]
To close the loop here: I added an (explicitly) untested (so no extra dependencies) example to the main coordinates page in #11529

**Pull request closed without merging**
